pkg/cluster/cluster_info_scraper.go

```diff
@@ -205,13 +206,32 @@ func (s *ClusterScraper) GetRunningAndReadyPodsOnNodes(nodeList []*api.Node) []*
 	return util.GetReadyPods(pods)
 }
 
+func (s *ClusterScraper) GetAllRunningAndReadyPods() ([]*api.Pod, error) {
+	pods := []*api.Pod{}
+	fieldSelector, err := fields.ParseSelector("status.phase=" + string(api.PodRunning))
+	if err != nil {
+		return pods, fmt.Errorf("failed to fetch all running and ready pods in cluster: %v", err)
+	}
+
+	pods, err = s.findSelectedPods(fieldSelector)
+	if err != nil {
+		return pods, fmt.Errorf("failed to fetch all running and ready pods in cluster: %v", err)
+	}
+	return util.GetReadyPods(pods), nil
+}
+
 // TODO, create a local pod, node cache to avoid too many API request.
 func (s *ClusterScraper) findRunningPodsOnNode(nodeName string) ([]*api.Pod, error) {
 	fieldSelector, err := fields.ParseSelector("spec.nodeName=" + nodeName + ",status.phase=" +
 		string(api.PodRunning))
 	if err != nil {
 		return nil, err
 	}
+
+	return s.findSelectedPods(fieldSelector)
+}
+
+func (s *ClusterScraper) findSelectedPods(fieldSelector fields.Selector) ([]*api.Pod, error) {
```

Comment:
Can you use the existing `GetPods()` function?

Reply:
Updated!